src/plugins/validation/validation.c

```diff
@@ -34,13 +38,34 @@ int elektraValidationSet (Plugin * handle ELEKTRA_UNUSED, KeySet * returned, Key
 
 	while ((cur = ksNext (returned)) != 0)
 	{
-		const Key * meta = keyGetMeta (cur, "check/validation");
+		const Key * regexMeta = keyGetMeta (cur, "check/validation");
+
+		if (!regexMeta) continue;
 
-		if (!meta) continue;
+		const Key * icaseValidation = keyGetMeta (cur, "check/validation/ignorecase");
+		const Key * lineValidation = keyGetMeta (cur, "check/validation/line");
+		const Key * wordValidation = keyGetMeta (cur, "check/validation/word");
```

Comment:
What happens if both line and word are set?
What about having "check/validation/match" or "check/validation/boundary" which contains line, word,.. (default or none).
This would also better fit to new features like "check/validation/type" that can contain basic, extended, perl, fixed